MetaMask/metamask-extension
# UX: Added account hiding in the Account List Menu

## **Description**

This PR add the functionality to hide an account in the account list

Note: This feature is behind a feature flag so we will add tests while removing the feature flag

## **Related issues**

Fixes: #21655 

## **Manual testing steps**

1. Run extension with `NETWORK_ACCOUNT_DND=1 yarn start`
2. Go to Account List Menu
3. Click on Account Options, hide account
4. See another section of hidden accounts in account list menu
5. In Hidden list of accounts, you can't pin the accounts. First you have to unhide the account and then can pin.
6. When we hide a pinned account, it should be unpinned first and then hidden

## **Screenshots/Recordings**


https://github.com/MetaMask/metamask-extension/assets/39872794/a20c4bbc-cdc8-4f21-ab1e-3f46b2842d59



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
